urllib3/response.py

```diff
@@ -56,14 +56,30 @@ class GzipDecoder(object):
 
     def __init__(self):
         self._obj = zlib.decompressobj(16 + zlib.MAX_WBITS)
+        self._state = 'first'
 
     def __getattr__(self, name):
         return getattr(self._obj, name)
 
     def decompress(self, data):
+        ret = binary_type()
+        if self._state == 'error':
+            return ret
```

Comment:
In the case of an error state, should we be returning nothing, or should we be raising an exception since there's no reasonable way to proceed?

Reply:
It should be returning an empty string.  The whole point of all these handling is to allow members other than the first to trigger errors and we will ignore them, so as to support files that have garbage at the end.  If this is not required, all the error handling can be removed and we can let the zlib error to propagate out.  I've added some comments in the code to explain this.